Source/SwiftLintFramework/Rules/TrailingCommaRule.swift

```diff
@@ -46,13 +50,21 @@ public struct TrailingCommaRule: ASTRule, ConfigurationProviderRule {
 
     public func validate(file: File, kind: SwiftExpressionKind,
                          dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        if let (index, reason) = violationIndexAndReason(in: file, kind: kind, dictionary: dictionary) {
+            return violations(file: file, byteOffset: index, reason: reason.rawValue)
+        } else {
+            return []
+        }
+    }
 
+    private func violationIndexAndReason(in file: File, kind: SwiftExpressionKind,
+                                         dictionary: [String: SourceKitRepresentable]) -> CommaRuleViolation? {
```

Comment:
I'm new to this codebase, but isn't returning empty array semantically the same as returning `nil`?

I mean, when returning optional, it first will be unwrapped and then checked with `isEmpty` anyways, so why not return empty array... I guess that was the rationale behind original design of this API.